src/Core/ConfigurationParser.re

```diff
@@ -205,6 +206,11 @@ let configurationParsers: list(configurationTuple) = [
     "files.exclude",
     (config, json) => {...config, filesExclude: parseStringList(json)},
   ),
+  (
+    "search.exclude",
+    (config, json) => { ...config, searchExclude: parseStringList(json)}
+
```

Comment:
This isn't documented very well in our developer docs, but we're starting to move away from this strategy of configuration - and instead specifying configuration per-feature.

A good example of this is here: https://github.com/onivim/oni2/blob/0f1bf1ea4b2da4590e7c5cd5d67d2cd994f09fd5/src/Feature/Zoom/Feature_Zoom.re#L93 (just recently added in #3021 )

Instead of implementing this here - we could add this to `Feature_Search`:
```
module Configuration = {
  open Config.Schema;

  let exclude = setting("search.exclude", list(string), ~default=[]);
};
```

We could follow a similar strategy for the `search.exclude` parameter, and add something like this in `Feature_Search`.

The reason for doing this, is that this will make it easier to plumb the configuration setting into the subscription runner here: https://github.com/onivim/oni2/blob/0f1bf1ea4b2da4590e7c5cd5d67d2cd994f09fd5/src/Feature/Search/Feature_Search.re#L212

Once we have that configuration setting in `Feature_Search`, we could modify the `let subscriptions = ...` function so that we can get the configuration setting:

```
let subscriptions = (~config: Oni_Core.Config.resolver, ~workingDirectory, ripgrep, dispatch) => {
  let searchExclude = Configuration.exclude.get(config);
  let search = query => {
    SearchSubscription.create(
      ~id="workspace-search",
      ~query,
      ~directory=workingDirectory,
      ~ripgrep, 
      ~searchExclude,
      ~onUpdate=items => dispatch(Update(items)),
      ~onCompleted=() => Complete,
      ~onError=msg => SearchError(msg),
    );
  };
  
   ...
};
```
(We'd also need to update `Feature_Search.rei` to add this new parameter).

The next thing we need to do, though, is get that `config` parameter - we could wire that in here: https://github.com/onivim/oni2/blob/0f1bf1ea4b2da4590e7c5cd5d67d2cd994f09fd5/src/Store/Features.re#L1883

something like:
```
(state: State.t, dispatch) => {
    ...
    let searchDispatch = msg => dispatch(Search(msg));
    let config = Selectors.configResolver(state);
    searchSubscriptions(~config, ~workingDirectory, searchDispatch, state.searchPane)
    |> SearchSubscriptionRunner.run(~dispatch=searchDispatch);
```

Hope that helps in terms of wiring up the configuration setting to the subscription. Let me know if you have questions!